### DISCUSSION TITLE
Question about Mode2 performance

### DISCUSSION THREAD

**Discussion by johanneszellinger** [Courseplay/courseplay#7260] [Q&A] [answered]
Didn't want to create a new issue (since I don't think this is a bug), but wanted to get informed. 

I have a simple setup:
-Krone BigX1150
-2 Tractors with Tippers. 
-Big not-uniformly shaped field (Field 39 of the "Somewhere in Thüringen / Irgendwo in Thüringen" map. 

I have just begun this map and FPS is locked to a constant 60FPS most of the time. However, during this harvest it seems that every time the tractors switch (one being full, the other empty and waiting) I get a huge fps drop to 15-19 FPS and the occasional "Tractor X is in traffic" icon. 

As soon as they have decided  what to do and the full tractor drives off, FPS is normal again. I have read some threads about the pathfinding  being the reason for this, however I wanted to know if that is the case here also. Question is, if a tractor is waiting at the field edge, does it calculate the whole path to the combine immediately before driving off? Or does it always only calculate the next waypoint and therefore update the path continuously? If the second case is true, than this doesn't really explain the behaviour, since as soon as a tractor drives off, my FPS are back to normal. 

Maybe I also need to test with a single tractor too, to make sure, when exactly FPS drops occur (and which machine is causing them).

**Comment by nitroSkirata** [comment, marked as answer]
The pathfinder runs before the tractor starts moving to the combine (generating the whole course to the combine) and before driving off when its full (generating the whole course to the first waypoint of the unload course).

So, you should get two FPS drops, one before the new tractor leaves for the combine and one when the old tractor moves away from the combine.

**Reply by johanneszellinger** [reply]
alright, this would explain the behavior. Thanks for the info :)
BTW: I haven't played FS in the last 2 years, but used CP quite extensively in FS17, and boi, the pathfinding has come a long way! Wow!

**Reply by Tensuko** [reply]
See also #7243 (and many other Issues or Discussions)
